### PR TITLE
Update API tests to use loss correctly

### PR DESCRIPTION
**Pull request by markdouthwaite**
This PR fixes a minor issue where the `test_coo_with_duplicate_entries` test in `test_api.py` does not loop over different loss functions as intended.

The PR also adds `.idea/*` (PyCharm project config) files to `.gitignore`.

### CODE REVIEW

**Comment by markdouthwaite** [comment]
I've also updated `test_movielens.py` to explicitly set `shuffle=True` to resolve a test failing due to (apparent) changes in the underlying Scikit-Learn API. Happy to break that into a separate PR if preferred.

**Comment by SimonCW** [comment]
Thank you @markdouthwaite, I'll request @maciejkula to review and merge.

File: tests/test_api.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def test_coo_with_duplicate_entries():
     mat.col = np.concatenate((mat.col, mat.col[:1000]))
 
     for loss in ("warp", "bpr", "warp-kos"):
-        model = LightFM(loss="warp")
+        model = LightFM(loss=loss)
```
Comment (maciejkula) [inline tests/test_api.py:73]: Great catch. Thanks!

File: tests/test_movielens.py [hunk 2]
```diff
@@ -704,7 +704,7 @@ def split(self, X, y=None, groups=None):
             for _ in range(self.n_splits):
                 yield idx, idx
 
-    cv = CV(n_splits=3, random_state=42)
+    cv = CV(n_splits=3, random_state=42, shuffle=True)
```
Comment (maciejkula) [inline tests/test_movielens.py:707]: We can't make this change as it would make tests non-deterministic.

Could you explain in more detail why this is needed? That will help us find another solution.

**Review by maciejkula — CHANGES_REQUESTED** [review]

File: tests/test_movielens.py (hunk 2 above)
Comment (maciejkula) [reply tests/test_movielens.py:707]: I verified what's going on, and this will not be non-deterministic. Let's merge.

**Review by maciejkula — APPROVED** [review]
